pkg/admissioncontroller/webhook/admission/resourcesize/handler.go

```diff
@@ -155,14 +218,56 @@ func isUnrestrictedUser(userInfo authenticationv1.UserInfo, subjects []rbacv1.Su
 	return userMatch(userInfo, subjects)
 }
 
-func findLimitForGVR(limits []admissioncontrollerconfigv1alpha1.ResourceLimit, gvr *metav1.GroupVersionResource) *resource.Quantity {
+func findRestrictionsForGVR(limits []admissioncontrollerconfigv1alpha1.ResourceLimit, gvr *metav1.GroupVersionResource) (*resource.Quantity, *int64) {
 	for _, limit := range limits {
 		size := limit.Size
+		count := limit.Count
 		if admissioncontrollerhelper.APIGroupMatches(limit, gvr.Group) &&
 			admissioncontrollerhelper.VersionMatches(limit, gvr.Version) &&
 			admissioncontrollerhelper.ResourceMatches(limit, gvr.Resource) {
-			return &size
+			return size, count
 		}
 	}
-	return nil
+	return nil, nil
+}
+
+func (h *Handler) getKindFromGVR(gvr *metav1.GroupVersionResource) (string, error) {
+	if h.RESTMapper == nil {
+		return "", errors.New("RESTMapper not initialized")
+	}
+
+	gvk, err := h.RESTMapper.KindFor(schema.GroupVersionResource{
+		Group:    gvr.Group,
+		Version:  gvr.Version,
+		Resource: gvr.Resource,
+	})
+	if err != nil {
+		return "", err
+	}
+	return gvk.Kind, nil
+}
+
+func (h *Handler) existingResourcesExceedLimit(_ admission.Request, gvr *metav1.GroupVersionResource, limit int64) (bool, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	// Get the proper Kind for this resource
+	kind, err := h.getKindFromGVR(gvr)
+	if err != nil {
+		return false, fmt.Errorf("failed to list resources for calculating object count: %w", err)
+	}
+
+	list := &metav1.PartialObjectMetadataList{}
+	list.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   gvr.Group,
+		Version: gvr.Version,
+		Kind:    kind + "List",
+	})
+
+	// List all resources of this type cluster-wide (since we only count non-namespaced resources)
+	if err := h.Client.List(ctx, list, client.Limit(limit+1)); err != nil {
```

Comment:
Nit: Would it make sense to use an `APIReader` (that is uncached)? Otherwise, if many objects were created at the same time, the GAC's cache might be stale and (unintentionally) allow exceeding the configured resource limit?